### PR TITLE
feat(AutoSuggest): Automatically close modal after selection

### PR DESCRIPTION
**Pull request by amir-zahedi**

### CODE REVIEW

File: lib/components/AutoSuggest/AutoSuggest.tsx [hunk 1]
```diff
@@ -328,6 +335,10 @@ const AutoSuggestInput = <PayloadType extends unknown>({
 								event.preventDefault();
 								if (typeof onChange === 'function')
 									onChange(suggestions[state.highlightIndex]);
+								if (typeof onSelection === 'function')
```
Comment (xavicosco) [inline lib/components/AutoSuggest/AutoSuggest.tsx]: Could these conditionals be done as a switch with multiple options (onChange, onSelection and so forth)?

**Comment by amir-zahedi** [comment]
@maraisr not sure how onChange can be used to close the portal when its triggered on each input change including key strokes

**Comment by maraisr** [comment]
@amir-zahedi as mentioned above.

The changes we need to make, in order to achieve the purpose of this PR are:

- Destruct the already passed in onChange handler.

https://github.com/autoguru-au/overdrive/blob/99814ba32770e57348a714fac6e6542cea20c694/lib/components/AutoSuggest/AutoSuggest.tsx#L91-L92

```diff
	onBlur: incomingOnBlur,
	onFocus: incomingOnFocus,
+   onChange: incomingOnChange,
```

- And `wrap` it, so that our callback runs before we forward to consumer. If consumer never passed in, the consumer callback will never fire. But ours will.

https://github.com/autoguru-au/overdrive/blob/99814ba32770e57348a714fac6e6542cea20c694/lib/components/AutoSuggest/AutoSuggest.tsx#L107-L108

```diff
        onFocus: wrapEvent(() => setIsFocused(true), incomingOnFocus),
		onBlur: wrapEvent(() => setIsFocused(false), incomingOnBlur),
+       onChange: wrapEvent(e => {
+           if (typeof e.payload !== 'undefined' && e.payload !== null) {
+               setIsFocused(false);
+           }
+       }, incomingOnChange),
```

Further that; we should also probably wrap the `props` const in a `useMemo`.

File: lib/components/AutoSuggest/AutoSuggest.tsx [hunk 2]
```diff
@@ -100,31 +101,33 @@ export const AutoSuggest = <PayloadType extends unknown>({
...
+				setIsFocused(false);
+			}
+
+			incomingOnChange(value);
+		},
 		itemRenderer,
 		onKeyDown,
 		onClick,
 		onFocus: wrapEvent(() => setIsFocused(true), incomingOnFocus),
-		onBlur: wrapEvent(() => setIsFocused(false), incomingOnBlur),
```
Comment (maraisr) [inline lib/components/AutoSuggest/AutoSuggest.tsx]: On desktop we want to close the model if the loose focus?

**Pull request merged**